### PR TITLE
stack.c: Add stack using array in C

### LINKED ISSUE #40: [DataStructure] Stack [C]

### PR DESCRIPTION
**Pull request by w4rb0y**
The code is modular and simple to understand for a newbie.

Stack is an ordered list of similar data type. Stack is a LIFO(Last in First out) structure or we can say FILO(First in Last out). push() function is used to insert new elements into the Stack and pop() function is used to remove an element from the stack.

Fixes #40 

**By submitting this pull request I confirm I've read and complied with the
below declarations.**

- [x] I have read the [Contribution guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [commit guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#write-good-commit-messages) for this project.
- [x] My code follows the [standard code structure](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#samples-code).
- [x] This pull request has a descriptive title. For example, `{Tag}: Add {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.
- [x] This pull request shall only be reviewed and merged once all the checks passes. No maintainer or supporter shall be obliged to review it before this condition is met.
- [x] I have mentioned the issue number correctly (with hyperlink) in this pull request description.

After you submit your pull request, **DO NOT** click the 'Update Branch' button.

### CODE REVIEW

File: stack/C/stackUsingArray.c [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	if(*top == n)
+		printf("\nStack is full. \n");
+	else
+		arr[++(*top)] = i;
+ }
+int pop(int *arr, int *top) {
+	if(*top == -1) {
+		printf("\n Stack empty!\n");
+		exit(1);
+	 }
```
Comment (sangamcse) [inline stack/C/stackUsingArray.c]: Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]

*Origin: CPPLintBear, Section: `all.cpplint`.*

**Comment by w4rb0y** [comment]
I tried my best to follow the strict indentation guidelines. But, I don't know if I'll ever be able to make though a single PR. I hope you'll fix the indentation problem.

Thank you.

File: stack/C/stackUsingArray.c [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	}
+	printf("\n");
+	// pop elements one by one.
+	for(int i = 0; i < n; i++) {
+		printf("%d ", pop(arr, &top));
+	}
+}
+void push(int *arr, int *top, int n, int i) {
+	if(*top == n)
+		printf("\nStack is full. \n");
```
Comment (sangamcse) [inline stack/C/stackUsingArray.c]: An else should appear on the same line as the preceding }  [whitespace/newline] [4]

*Origin: CPPLintBear, Section: `all.cpplint`.*

File: stack/C/stackUsingArray.c (hunk 1 above)
Comment (w4rb0y) [reply stack/C/stackUsingArray.c]: I tried fixing this in f487f893db0005853d256fae370cc15e69c4bd5f commit.

**Comment by w4rb0y** [comment]
I'm unable to resolve this last indentation error. @sangamcse  help me with this.

File: stack/C/stackUsingArray.c [hunk 3]
```diff
@@ -0,0 +1,38 @@
+#include<stdio.h>
+#include<stdlib.h>
+
+/*Using variable i to push elements then pop and print the elements. */
+
+void main() {
```
Comment (sangamcse) [inline stack/C/stackUsingArray.c]: This needs to be `int main(...)`

Comment (w4rb0y) [reply stack/C/stackUsingArray.c]: done.

**Comment by sangamcse** [comment]
Cool. Now you need to squash your commits into one. You can take help from https://github.com/NITSkmOS/Algorithms/blob/master/GIT_BASICS.md

**Pull request merged**